app/graphql/types/installment.rb

```diff
@@ -0,0 +1,31 @@
+class Types::Installment < Types::BaseObject
+  implements Types::Interface::WithTimestamps
+
+  description ''
+
+  field :id, ID, null: false
+
+  field :release_position, Integer,
+    null: true,
+    description: ''
+
+  field :chronological_position, Integer,
+    null: true,
+    description: ''
```

Comment:
We probably want to hide the actual order values and make it look contiguous to clients, so they don't have to think about our database implementation details. ranked_model provides a `release_position_rank` method which does this, but does it by `where(x < y).count` which means a separate query each time.

Since that would have absolute garbage-ass performance, we'll probably want to add a window function to our query to get (effectively) `ROW_NUMBER() OVER (PARTITION BY #{table[@find_by]} ORDER BY release_position ASC)`

So we probably want to add an escape hatch to FancyLoader to allow modifying the `base_query` before it generates the rest of the query. I imagine it could look something like this:

```ruby
class Loaders::InstallmentsLoader < Loaders::FancyLoader
  from Installment
  modify_query ->(query) {
    release_rank = RankQueryGenerator.new(:release_position, partition).arel
    query.project(release_rank)
  }

  sort :release_position
  sort :chronological_position
end
```

Then RankQueryGenerator would just do a bunch of Arel stuff to build up the partition statement like the existing `Loaders::FancyLoader::QueryGenerator`

Reply:
interesting. I'll see if I can manage that.